Add Juliaup docs (#50477)
JuliaLang/julia: doc/src/manual/installation.md

Juliaup with this installation method. The 64 bit version of the MSI installer
can be downloaded from [here](https://install.julialang.org/Julia-x64.msi) and
the 32 bit version from [here](https://install.julialang.org/Julia-x86.msi).

 By default the install will be a per-user install that does not require
 elevation. You can also do a system install by running the following command
 from a shell:

```
msiexec /i <PATH_TO_JULIA_MSI> ALLUSERS=1
```

### [Homebrew](https://brew.sh) (Mac and Linux)

On systems with brew, you can install Julia by running
```
brew install juliaup
```
in a shell. Note that you will have to update Juliaup with standard brew
commands.

### [Arch Linux - AUR](https://aur.archlinux.org/packages/juliaup/) (Linux)

On Arch Linux, Juliaup is available [in the Arch User Repository (AUR)](https://aur.archlinux.org/packages/juliaup/).

### [openSUSE Tumbleweed](https://get.opensuse.org/tumbleweed/) (Linux)

On openSUSE Tumbleweed, you can install Julia by running

```sh
zypper install juliaup
```
in a shell with root privileges.

### [cargo](https://crates.io/crates/juliaup/) (Windows, Mac and Linux)

> To install Julia via Rust's cargo, run:
> 
> ```sh
> cargo install juliaup
> ```

Reviewer: The text says: "To install Julia via Rust's cargo, run:" but then gives a command to install `juliaup`. A non-experienced reader at this point might expect Julia to already be installed.

How about this:
````suggestion
To install Juliaup via Rust's cargo, run:

```sh
cargo install juliaup
```
Afterwards run `juliaup` -- you may pass any of the command line arguments described above.
````

I guess the same also applies to the OpenSUSE and (home)brew sections -- and to a degree also the Arch one, which correctly states that it describes how to install Juliaup, but the user reading this wants to install Julia.

Author: I added some language at the top of the alternative methods section, I think that might be good enough?

In any case, I think we should just merge this at this point, and then folks can open follow up PRs that refine the lanugage.

# [Installation](@id man-installation)

There are many ways to install Julia. The following sections highlight the recommended method for each of the main supported platforms, and then present alternative ways that might be useful in specialized situations.

> Julia switched the default installation recommendation to a solution based on Juliaup in the summer of 2023. If you installed Julia previously with a method that is _not_ based on Juliaup and want to switch your system to an installation that is based on Juliaup, we recommend that you uninstall all previous Julia versions, ensure that you remove anything Julia related from your `PATH` variable and then install Julia with one of the methods described below.

Reviewer: Is the information about when this change happened really helpful? Most users wouldn't care, would they? They just want a straight answer "what should I do to install Julia?", I'd reckon (experts who want choices can read on / look elsewhere). So how about this:
```suggestion
The current installation recommendation is a solution based on Juliaup. If you installed Julia previously with a method that is _not_ based on Juliaup and want to switch your system to an installation that is based on Juliaup, we recommend that you uninstall all previous Julia versions, ensure that you remove anything Julia related from your `PATH` variable and then install Julia with one of the methods described below.
```
Though I find it somewhat confusing that it talks about "methods/solutions *based on* Juliaup" instead of just saying "installed using Juliaup". Is there a reason for that? I get that it is technically more accurate, but would the average user care?

(For that matter, I am not starting to wonder: would many users even *know* whether they used Juliaup before? If so, they might read this and wonder "does this apply to me? how do I find out?". I am not making this up, I've had to help dozens of academics installing Julia and it's amazing what all can go wrong and be misunderstood, so much that they also often never upgrade from whatever version we installed -- this is also why I am quite excited about Juliaup, because I am hoping its update capabilities will get more of my colleagues to use recent Julia versions. Anyway, I still might be worrying to much here -- time will tell...)

Reviewer: Also, maybe `Juliaup` could be a link to documentation for `Juliaup` (well... I guess that would require that to be created first ;-) but in the meantime perhaps at least a link to https://github.com/JuliaLang/juliaup would be helpful.

Author: I've adopted your language for the first recommendation (once I push). I'm still keeping the "based on" because at least for Windows users the term "Juliaup" never shows up when they install, so I'm afraid that a Windows user might be confused by "Installed using Juliaup". But we can iterate, I thought maybe we'll merge this PR rather sooner than later, and then we can refine the language?

```

Here `<ARGS>` should be replaced with one or more of the following arguments:
- `--yes` (or `-y`): Run the installer in a non-interactive mode. All configuration values use their default or a value supplied as a command line argument.
- `--default-channel=<NAME>`: Configure the default Juliaup channel. For example `--default-channel lts` would install the `lts` channel and configure it as the default.
- `--add-to-path=<yes|no>`: Configure whether Julia should be added to the `PATH` environment variable. Valid values are `yes` (default) and `no`.
- `--background-selfupdate=<SECONDS>`: Configure an optional CRON job that auto-updates Juliaup if `<SECONDS>` has a value larger than 0. The actual value controls how often the CRON job will run to check for a new Juliaup version in seconds. The default value is 0, i.e. no CRON job will be created.
- `--startup-selfupdate=<MINUTES>`: Configure how often Julia will check for new versions of Juliaup when Julia is started. The default is every 1440 minutes.
- `-p=<PATH>` (or `--path`): Configure where the Julia and Juliaup binaries are installed. The default is `~/.juliaup`.

## Alternative installation methods

> Note that we recommend the following methods _only_ if none of the installation methods described above work for your system.

Reviewer: This whole document is catered towards personal use of Julia, which of course is often nice.

In university labs, though, admins often want to install Julia in a central location. Should we give a pointer here or somewhere about that? That would also cover people who don't like to pipe output of `curl` into `sh` for security reasons.

This could be an additional section at the end of this document. (I am not suggesting this idea should hold back this PR -- it definitely shouldn't! Just thinking out loud.

Author: Yes, lets handle that in a follow up PR! Or even better, fix Juliaup so that it works in those scenarios ;)

### App Installer (Windows)

If the Windows Store is blocked on a system, we have an alternative [MSIX App Installer](https://learn.microsoft.com/en-us/windows/msix/app-installer/app-installer-file-overview) based setup. To use the App Installer version, download [this](https://install.julialang.org/Julia.appinstaller) file and open it by double clicking on it.

### MSI Installer (Windows)

If neither the Windows Store nor the App Installer version work on your Windows system, you can download a MSI installer from XXX. Note that this installation methods comes with serious limitations and is generally not recommended unless no other method works. For example, there is no automatic update mechanism for Juliaup with this installation method.

### [Homebrew](https://brew.sh) (Mac and Linux)

On systems with brew, you can install Julia by running
```
brew install juliaup
```
in a shell. Note that you will have to update Juliaup with standard brew commands.

> 

Reviewer: ```suggestion

```

# [Installation](@id man-installation)

There are many ways to install Julia. The following sections highlight the recommended method for each of the main supported platforms, and then present alternative ways that might be useful in specialized situations.

> The current installation recommendation is a solution based on Juliaup. If you installed Julia previously with a method that is _not_ based on Juliaup and want to switch your system to an installation that is based on Juliaup, we recommend that you uninstall all previous Julia versions, ensure that you remove anything Julia related from your `PATH` variable and then install Julia with one of the methods described below.

Author: I checked, and line wraps seem to be the norm, so I just added them.

 By default the install will be a per-user install that does not require elevation. You can also do a system install by running the following command from a shell:

```
msiexec /i <PATH_TO_JULIA_MSI> ALLUSERS=1
```

### [Homebrew](https://brew.sh) (Mac and Linux)

On systems with brew, you can install Julia by running
```
brew install juliaup
```
in a shell. Note that you will have to update Juliaup with standard brew commands.

### [Arch Linux - AUR](https://aur.archlinux.org/packages/juliaup/) (Linux)

On Arch Linux, Juliaup is available [in the Arch User Repository (AUR)](https://aur.archlinux.org/packages/juliaup/).

### [openSUSE Tumbleweed](https://get.opensuse.org/tumbleweed/) (Linux)

On openSUSE Tumbleweed, you can install Julia by running

```sh
zypper install juliaup
```
in a shell with root privileges.

### [cargo](https://crates.io/crates/juliaup/) (Windows, Mac and Linux)

To install Julia via Rust's cargo, run:

```sh

> cargo install juliaup

Reviewer: Does this do more than just installing `juliaup`, i.e., does it run `juliaup` as well? Or is this another step the user has to take?

Author: It just installs Juliaup, nothing else.
